src/main/java/seedu/address/model/ItemModelManager.java

```diff
@@ -213,6 +219,12 @@ public void addItem (Item item) {
         addToSeparateList(item);
     }
 
+    public void addItem(int targetIndex, Item item){
+        visualList.add(targetIndex, item);
+        itemStorage.add(targetIndex, item);
+        addToSeparateList(targetIndex, item);
```

Comment:
I think that the targetIndex is specific to the list shown when the item was deleted, I'm not sure if it the same index applies for all the lists?

Reply:
oh i did not think of that...opinions on creating a model that holds item and its index in a list?